### PR TITLE
[MAINT] Add CITATION.cff file

### PR DESCRIPTION
**Pull request by tsalo**
Closes None.

## Changes proposed in this pull request
- Add a CITATION.cff file with information from the Zenodo page.
- Add a documentation page describing the release process.

## Documentation that should be reviewed

The new developers guide page.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    family-names: Sydnor
+    affiliation: University of Pennsylvania
+    orcid: 'https://orcid.org/0000-0002-8640-668X'
+  - given-names: Sydney
+    family-names: Covitz
+    affiliation: University of Pennsylvania
+    orcid: 'https://orcid.org/0000-0002-7430-4125'
+identifiers:
+  - type: doi
+    value: 10.5281/zenodo.5139942
```
Comment (tsalo) [inline CITATION.cff:54]: The one thing I don't have a solution for is updating the Zenodo DOI. The version-specific DOI is created when the GitHub release is published, but we would need to update the Zenodo DOI in the CITATION.cff file before the release if we want folks to cite version-specific DOIs...

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    affiliation: University of Minnesota
+    orcid: 'https://orcid.org/0000-0003-3030-6580'
+  - given-names: Valerie
+    family-names: Sydnor
+    affiliation: University of Pennsylvania
+    orcid: 'https://orcid.org/0000-0002-8640-668X'
+  - given-names: Sydney
+    family-names: Covitz
+    affiliation: University of Pennsylvania
+    orcid: 'https://orcid.org/0000-0002-7430-4125'
```
Comment (tsalo) [inline CITATION.cff:51]: Should Ted and/or Damien be added to the end?

Comment (kahinimehta) [reply CITATION.cff:51]: I believe so? @mattcieslak

**Review by kahinimehta — APPROVED** [review]

File: CITATION.cff (hunk 2 above)
Comment (tsalo) [reply CITATION.cff:51]: Sounds good. Does it look like anyone else is missing?

Comment (tsalo) [reply CITATION.cff:51]: Per today's DCAN call, this is something we can bring up in Friday's call.

Comment (tsalo) [reply CITATION.cff:51]: We didn't get around to it, so I'm going to merge this and we can update it in the future if necessary.

**Pull request merged**